tests/test_core.py

```diff
@@ -627,16 +627,18 @@ def __test(signal, order, true_coeffs, est_coeffs):
                    test_data['est_coeffs'][i])
 
 
-def test_lpc_simple():
+@pytest.mark.parametrize('dtype', [np.float32, np.float64])
+def test_lpc_simple(dtype):
     srand()
 
     n = 5000
-    est_a = np.zeros((n, 6))
-    truth_a = [1, 0.5, 0.4, 0.3, 0.2, 0.1]
+    est_a = np.zeros((n, 6), dtype=dtype)
+    truth_a = np.array([1, 0.5, 0.4, 0.3, 0.2, 0.1], dtype=dtype)
     for i in range(n):
-        noise = np.random.randn(1000)
+        noise = np.random.randn(1000).astype(dtype)
         filtered = scipy.signal.lfilter([1], truth_a, noise)
         est_a[i, :] = librosa.lpc(filtered, 5)
```

Comment:
`lfilter` is returning `filtered` as having `dtype=np.float64`, and this test is passing.

Changing the lpc call to `librosa.lpc(filtered.astype(dtype), 5)` exposes the type error originally reported.